SonarRunner.Shim/SonarRunner.Wrapper.cs

```diff
@@ -82,25 +90,56 @@ private static bool ExecuteJavaRunner(ILogger logger, string exeFileName, string
 
             string args = string.Format(System.Globalization.CultureInfo.InvariantCulture,
                 "-Dproject.settings=\"{0}\"", propertiesFileName);
-            
+
             logger.LogMessage(Resources.DIAG_CallingSonarRunner);
 
+            string sonarRunnerOptsValue = GetSonarRunnerOptsValue(logger);
+
             ProcessRunner runner = new ProcessRunner();
-            bool success = runner.Execute(exeFileName, args, Path.GetDirectoryName(exeFileName), SonarRunnerTimeoutInMs, logger);
+            bool success = runner.Execute(
+                exeFileName,
+                args,
+                Path.GetDirectoryName(exeFileName),
+                Timeout.Infinite,
+                !String.IsNullOrEmpty(sonarRunnerOptsValue) ? new Dictionary<string, string>() { { SonarRunnerOptsVariableName, sonarRunnerOptsValue } } : null,
+                logger);
+
             success = success && !runner.ErrorsLogged;
 
             if (success)
             {
                 logger.LogMessage(Resources.DIAG_SonarRunnerCompleted);
             }
-			else
+            else
             {
-				// TODO: should be kill the process or leave it? Could we corrupt the data on the server if we kill the process?
+                // TODO: should be kill the process or leave it? Could we corrupt the data on the server if we kill the process?
                 logger.LogError(Resources.ERR_SonarRunnerExecutionFailed);
             }
             return success;
         }
 
+        /// <summary>
+        /// Get the value of the SONAR_RUNNER_OPTS variable that controls the amount of memory available to the JDK so that the sonar-runner doesn't 
+        /// hit OutOfMemory exceptions. If no env variable with this name is defined at machine or user level then use the process one is used. 
+        /// Bar that, a default value is used. 
+        /// </summary>
+        /// <returns></returns>
```

Comment:
Remove the empty <returns>.
The comment shouldn't mention machine/user/process level; it should just say that a default will be used if the variable isn't already set.